torchft/local_sgd.py

```diff
@@ -252,8 +286,10 @@ def _perform_sync(self) -> None:
         """
         # Set the .grad field of each parameter to its pseudogradient
         for name, p in self._model.named_parameters():
-            pseudogradient = p.data - self.original_parameters[name]
-            p.grad = pseudogradient
+            local_param = extract_local_tensor(p.data)
+            pseudogradient = local_param - self.original_parameters[name]
+            # TODO: is there something other than _local_tensor?
+            p._local_tensor.grad = pseudogradient
```

Comment:
cc @XilunWu, is there a better way to update the gradient of a `DTensor`?

If I try to set p.grad directly I get an error like:
```
    File "/data/users/howardhuang/torchft/torchft/local_sgd.py", line 293, in _perform_sync
      p.grad = pseudogradient
  RuntimeError: attempting to assign a gradient of size '[564, 256]' to a tensor of size '[2256, 256]'. Please ensure that the gradient and the tensor are the same size
```